### PR TITLE
Add backstage tracking and docs

### PR DESCRIPTION
**Pull request by dasconnor**
Currently just added our readme as a docs site index.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+site_name: avalanche-cli
+site_description: |
+  Avalanche-CLI is a command-line interface for building, deploying, and maintaining Avalanche
+  Subnets. It can be used to prototype and test Subnets, as well as to deploy them to the
+  Avalanche network. Avalanche-CLI is available for Linux and Mac, and is open source and
+  available on GitHub.
```
Comment (felipemadero) [inline mkdocs.yml:6]: duplicate mention of available

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+site_name: avalanche-cli
+site_description: |
+  Avalanche-CLI is a command-line interface for building, deploying, and maintaining Avalanche
+  Subnets. It can be used to prototype and test Subnets, as well as to deploy them to the
```
Comment (felipemadero) [inline mkdocs.yml:4]: would prefer the inverse order, or make it very clear that it is a very good option to deploy subnets to avalanche (it seems from the paragraph that its main goal is prototype and testing? and secondary, deploy)

File: catalog-info.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: avalanche-cli
+  description: |
+    Avalanche-CLI is a command-line interface for building, deploying, and maintaining Avalanche
```
Comment (felipemadero) [inline catalog-info.yaml:6]: do we need duplication of this?

Comment (felipemadero) [reply catalog-info.yaml:6]: can we just have on root level yml? (sorry if ignorant on this kind of conf)

File: catalog-info.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    available on GitHub.
+  annotations:
+    backstage.io/techdocs-ref: dir:.
+    github.com/project-slug: ava-labs/avalanche-cli
+spec:
+  type: cli
+  owner: platform-tooling
+  lifecycle: production
+  dependsOn:
+    - component:avalanchego
```
Comment (felipemadero) [inline catalog-info.yaml:18]: is this relevant? were are the other components specs to be found by the tools?

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
+# Avalanche-CLI
```
Comment (felipemadero) [inline docs/index.md:1]: can we avoid duplicating the README?

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+Restart your local deployment (from where you left off) with:
+
+```bash
+avalanche network start
+```
+
+## Notable Features
+
+- Creation of Subnet-EVM, and custom virtual machine subnet configurations
```
Comment (felipemadero) [inline docs/index.md:60]: we need to update this a little

File: docs/index.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+`subnet deploy` can be used on top of it. Notice that you need to continue giving `--snapshot-name` flag to those commands if you
+continue saving/restoring to it, if not, `default snapshot will be used`.
+
+### Snapshots dir
+
+- `~/.avalanche-cli/snapshot` will contain all saved snapshots, which can for example be used to pass work around
+
+## Detailed Usage
+
+More detailed information on how to use Avalanche CLI can be found at [here](https://docs.avax.network/subnets/create-a-local-subnet#subnet).
```
Comment (felipemadero) [inline docs/index.md:192]: is this the right link

**Review by felipemadero — COMMENTED** [review]
as a general note, should avoid duplication of info (ie READMEx2, yml package desc x2)
as I am not used to the doc generator, ignore if this duplication is needed or can be solved
I do really like this idea of making a doc tree for our tool doc, this is also needed by ANR 
last node: README needs update

File: mkdocs.yml (hunk 1 above)
Comment (dasconnor) [reply mkdocs.yml]: fixed

File: mkdocs.yml (hunk 2 above)
Comment (dasconnor) [reply mkdocs.yml]: rewrote

File: catalog-info.yaml (hunk 3 above)
Comment (dasconnor) [reply catalog-info.yaml:6]: sadly we have to duplicate here

File: catalog-info.yaml (hunk 4 above)
Comment (dasconnor) [reply catalog-info.yaml:18]: yeah, I want to start tracking all these dependencies

File: docs/index.md (hunk 5 above)
Comment (dasconnor) [reply docs/index.md]: Done! Turned the index into a symlink

File: docs/index.md (hunk 6 above)
Comment (dasconnor) [reply docs/index.md]: Created #898 for this

File: docs/index.md (hunk 7 above)
Comment (dasconnor) [reply docs/index.md]: yep, link is still good

**Review by felipemadero — APPROVED** [review]
LGTM

**Pull request merged**